universal-application-tool-0.0.1/app/forms/QuestionForm.java

```diff
@@ -63,12 +63,18 @@ public void setQuestionDescription(String questionDescription) {
     this.questionDescription = checkNotNull(questionDescription);
   }
 
-  public Path getQuestionPath() {
-    return questionPath;
+  public Path getQuestionParentPath() {
+    return questionParentPath;
   }
 
-  public void setQuestionPath(String questionPath) {
-    this.questionPath = Path.create(checkNotNull(questionPath));
+  public void setQuestionParentPath(String questionParentPath) {
+    this.questionParentPath = Path.create(checkNotNull(questionParentPath));
+  }
+
+  public Path getQuestionPath() {
+    String questionNameFormattedForPath =
+        questionName.replaceAll("\\s", "_").replaceAll("[^a-zA-Z_]", "");
+    return questionParentPath.with(questionNameFormattedForPath);
```

Comment:
nit: feel free to disagree, but I feel like `with` could be renamed to `append`

Reply:
I'm gonna use "join", from python's `os.path.join` and node's `path.join`.